1875_Mp3_Player_Clone.js

```diff
@@ -0,0 +1,316 @@
+    let querySplitter = query.toLowerCase().split(' ');
+    let flagTrue;
+
+    for(let i in wordSplitter) //runs each word at a time
+    {
+      for(let t in querySplitter){ //runs all query/ies tests on each word at a time. to che all options.
+        if(wordSplitter[i].includes(querySplitter[t])) flagTrue = true; //if query included in word then flagtrue is true which means its included...
+      } 
+   } 
+   if(flagTrue !== undefined && flagTrue === true) return flagTrue; //if the flagtrue got assigned a value(in this case always true). return true(means included)
+   return false; // else return false.(no matter the case)
+ }
+
+
+
+  function searchByDuration(duration) {
+
+    let seconds = mmtoss(duration);                                                                  
+    let songsDurationsArr =[], playlistDurationsArr = [], minPlaylistArr = [], minSongArr = [];
+
+      playlistDurationArr = player.playlists.map(playlist => mmtoss(playlistDuration(playlist.id))); //inserts & converts song duration mm:ss form to ss form to an array representing each playlit duration!
+      
+      songsDurationsArr = player.songs.map(song => mmtoss(songDuration(song.id))); //inserts & converts song duration mm:ss form to ss form to ss form to an array representing each song duration!
+      
+    //playlistDuration.forEach(element => mmtoss(element)); //converts playlist duration mm:ss form to ss form!
+    //songDuration.forEach(element => mmtoss(element)); //converts song duration mm:ss form to ss form!
+
+    minPlaylistArr = playlistDurationArr.map(length => Math.abs(seconds - length));
+     //minPlaylistArr = Math.min(...minPlaylistArr);
+    minSongArr = songsDurationsArr.map(length => Math.abs(seconds - length));
+    //minSongArr = Math.min(...minSongArr);
+    
+
+    if(Math.min(...minPlaylistArr) > Math.min(...minSongArr)) return findSongByDuration(Math.min(...minSongArr)+seconds); //output the original song by his original duration by adding the missing "seconds" which we substract earlier!
+    return findPlaylistByDuration(Math.min(...minPlaylistArr)+seconds); //output the original playlist by his original duration by adding the missing "seconds" which we substract earlier!
+  } 
+
+
+  function songDuration(id){ // gets song id and return its duration in mm:ss
+    if(getSongIdIndex(id) === undefined) throw("song id not found in list");
+    //let durArr= player.songs.filter(element => (element.id === id));
+    //let songDur = durArr[0].duration;
+    let songDur = player.songs[getSongIdIndex(id)].duration 
+    return sstomm(songDur);
+  }
+
+  function findSongByDuration(songDur){ //get song duration (IN SECONDS!) and return the song which its duration is equal to the inserted one! if there are more than 1 song same duration then it outputs all songs with this duration!
+    let songArr = []; //array of all songs with the given duration!
+   /* for(let i=0; i<player["songs"].length; i++)
```

Comment:
remove unnecessary code

Reply:
yep, it just for me so i can learn from my mistakes , in the future  and prevent them